master/internal/task/all_gather.go

```diff
@@ -98,7 +98,10 @@ func (g *allGather) unwatch(msg UnwatchAllGather) {
 }
 
 func (g *allGather) checkTimeout(msg allGatherTimeout) error {
-	if g.id == msg.id {
```

Comment:
why reverse this `==`?  Is that secretly fixing a separate bug or was the change an accident?

Reply:
accident, good catch